### PR TITLE
PA-635 add frontend tests for SPL Reports page and related functionality

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: frontend/src/features/splReports/components/SplReportLayout.scss [hunk 1]
```diff
@@ -61,7 +61,6 @@
 
   .tbody {
     max-height: calc(100vh - #{$footer-height} - #{$header-height} - #{$navbar-height} - 110px);
-    background-color: red;
```
Comment (devinleighsmith) [inline frontend/src/features/splReports/components/SplReportLayout.scss:64]: How this got in here I'll never understand.

File: frontend/src/features/splReports/components/ReportControls.tsx [hunk 2]
```diff
@@ -147,7 +146,12 @@ const ReportControls: React.FunctionComponent<IReportControlsProps> = ({
                 <Button
                   className="h-75 mr-auto"
                   type="submit"
-                  disabled={(!dirty && originalReport?.to === currentReport?.to) || !values.id}
+                  disabled={
+                    (!dirty &&
+                      originalReport?.to === currentReport?.to &&
+                      originalReport?.from === currentReport?.from) ||
```
Comment (devinleighsmith) [inline frontend/src/features/splReports/components/ReportControls.tsx:152]: from has the same issue as to.

File: frontend/src/components/common/form/Check.tsx [hunk 3]
```diff
@@ -85,7 +85,6 @@ export const Check: React.FC<CheckProps> = ({
             required={required}
             disabled={disabled}
             custom={custom}
-            defaultChecked={!!checked}
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/Check.tsx:88]: Having both checked and defaultchecked throws a warning, controlled components should be using the checked prop so I've switched to that.

File: frontend/src/components/common/form/Input.tsx [hunk 4]
```diff
@@ -94,30 +94,29 @@ export const Input: React.FC<InputProps> = ({
           {!!tooltip && <TooltipIcon toolTipId={`${field}-tooltip`} toolTip={tooltip} />}
         </Form.Label>
       )}
-      <span className={className}>
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/Input.tsx:97]: As far as I can see there is no reason for this span to be here, I've had a look through the pages affected by removing this and there doesn't appear to be a difference. Removing this:
simplifies the dom
makes findByLabelText work

File: frontend/src/features/splReports/containers/SplReportContainer.tsx [hunk 5]
```diff
@@ -101,19 +95,6 @@ const SplReportContainer: React.FunctionComponent<ISplReportContainerProps> = ()
     setCurrentReport(report);
     setSnapshots(data);
   };
-  const exportReport = (report: IReport, accept: 'csv' | 'excel') => {
```
Comment (devinleighsmith) [inline frontend/src/features/splReports/containers/SplReportContainer.tsx:104]: This makes more sense to be in the api hook.

File: frontend/src/features/splReports/hooks/useRouterReport.ts [hunk 6]
```diff
@@ -34,17 +34,16 @@ export const useRouterReport = ({
...
 
   //When this hook loads, override the value of the filter with the search params. Should run once as originalSearch should never change.
   useDeepCompareEffect(() => {
-    if (reports?.length) {
+    if (reports?.length && currentReport?.id === undefined) {
       const filterFromParams = paramsToObject(originalSearch);
       if (filterFromParams.reportId) {
         const report = _.find(reports, { id: +filterFromParams.reportId });
         setCurrentReport(report ?? reports[0]);
-        setOriginalSearch('');
```
Comment (devinleighsmith) [inline frontend/src/features/splReports/hooks/useRouterReport.ts:47]: setting original search like this was causing an unnecessary call of this useEffect. The change has the same effect of only setting the report if unset, but without this extra evaluation.

**Comment by devinleighsmith** [comment]
You may want to skip that majority of the snapshots on the code review.

**Review by jdtoombs — APPROVED** [review]

File: frontend/package.json [hunk 7]
```diff
@@ -69,6 +69,7 @@
     "@types/lodash": "4.14.149",
     "@types/mui-datatables": "2.13.7",
     "@types/node": "12.12.17",
+    "@types/pretty": "^2.0.0",
```
Comment (Fosol) [inline frontend/package.json:72]: Should this be a specific version

**Review by Fosol — APPROVED** [review]
That should improve our coverage

File: frontend/package.json (hunk 7 above)
Comment (devinleighsmith) [reply frontend/package.json:72]: It should be, thanks.

**Comment by devinleighsmith** [comment]
snyk died for some reason. I'll force push at some point to force it to run again.

**Pull request merged**